person.rb

```diff
@@ -1,23 +1,56 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
 class Person
   attr_reader :id
   attr_accessor :name, :age, :parent_permission
 
-  def initialize(age:, name: 'unknown', parent_permission: true)
+  def initialize(*args)
     @id = Random.rand(1..1000)
-    @name = name
-    @age = age
-    @parent_permission = parent_permission
+    @age = args[0]
+    @name = args[1] || 'unknown'
+    @parent_permission = args[2].nil? ? true : args[2]
   end
 
   def can_use_services?
     of_age? || @parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
   private
 
   def of_age?
     @age >= 18
   end
 end
 
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
 
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name()
+    @nameable.correct_name.upcase
```

Comment:
- [ ] Please the method that convert the first letter of a work to uppercase is the `capitalize` method, something like so: 
```
  @nameable.correct_name.capitalize
```
Please kindly fix it.